src/Symfony/Component/Security/Http/Event/UserHasChangedEvent.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Event;
+
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Contracts\EventDispatcher\Event;
+
+/**
+ * @author Hamza Amrouche <hamza.simperfit@gmail.com>
+ */
+class UserHasChangedEvent extends Event
+{
+    private $token;
+    private $user;
+    private $userNotFoundByProvider;
+    private $userDeauthenticated;
+
+    public function __construct(TokenInterface $token, UserInterface $user, bool $userDeauthenticated, bool $userNotFoundByProvider)
+    {
+        $this->token = $token;
+        $this->user = $user;
+        $this->userDeauthenticated = $userDeauthenticated;
+        $this->userNotFoundByProvider = $userNotFoundByProvider;
+    }
+
+    /**
+     * @return TokenInterface The token used to refresh the user
+     */
+    public function getToken(): TokenInterface
+    {
+        return $this->token;
+    }
+
+    /**
+     * @return UserInterface The first given user
+     */
+    public function getUser(): UserInterface
+    {
+        return $this->user;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isUserNotFoundByProvider(): bool
```

Comment:
I missed that change, it makes no sense to me. 
Deauthentication means  by definition that a user has been found. At this point `$userNotFoundByProvider` would only mean that a previous provider failed to load the user (a different provider than the one which deauthenticated the token).  The event class should be reverted to its previous state with only `getUser()` and `getToken()` IMO.